zeus/device/cpu/rapl.py

```diff
@@ -0,0 +1,180 @@
+
+    def __init__(self, message: str) -> None:
+        """Initialize Zeus Exception."""
+        super().__init__(message)
+
+
+class ZeusRAPLFileInitError(ZeusBaseCPUError):
+    """Zeus CPU exception class Wrapper for RAPL file initialization error on CPU."""
+
+    def __init__(self, message: str) -> None:
+        """Initialize Zeus Exception."""
+        super().__init__(message)
+
+
+class RAPLFile:
+    """RAPL File class for each RAPL file.
+
+    This class defines the interface for interacting with a RAPL file for a package. A package can
+    be a CPU or DRAM
+    """
+
+    def __init__(self, path: str) -> None:
+        """Initialize RAPL file object. Each RAPL file object manages one energy_uj file."""
+        self.path = path
+        self.energy_uj_path = os.path.join(path, "energy_uj")
+        try:
+            with open(os.path.join(path, "name"), "r") as name_file:
+                self.name: str = name_file.read().strip()
+        except FileNotFoundError as err:
+            raise ZeusRAPLFileInitError("Error reading package name") from err
+        try:
+            with open(self.energy_uj_path) as energy_file:
+                self.last_energy = int(float(energy_file.read().strip()) / 10**3)
+        except FileNotFoundError as err:
+            raise ZeusRAPLFileInitError("Error reading package energy") from err
+        try:
+            with open(
+                os.path.join(path, "max_energy_range_uj"), "r"
+            ) as max_energy_file:
+                self.max_energy_range_uj = int(
+                    float(max_energy_file.read().strip()) / 10**3
+                )
+        except FileNotFoundError as err:
+            raise ZeusRAPLFileInitError(
+                "Error reading package max energy range"
+            ) from err
+
+    def __str__(self) -> str:
+        """Return a string representation of the RAPL file object."""
+        return f"Path: {self.path}\nEnergy_uj_path: {self.energy_uj_path}\nName: {self.name}\
+        \nLast_energy: {self.last_energy}\nMax_energy: {self.max_energy_range_uj}"
+
+    def _read(self) -> int:
+        """Read the current energy value from the energy_uj file.
+
+        Returns:
+            int: The current energy value in millijoules.
+        """
+        with open(self.energy_uj_path) as energy_file:
+            self.last_energy = int(float(energy_file.read().strip()) / 10**3)
+        return self.last_energy
+
+    def _read_delta(self) -> int:
+        """Read the delta energy value since the last read.
+
+        Returns:
+            int: The delta energy value in millijoules.
+        """
+        last_energy: int = self.last_energy
+        new_energy: int = self._read()
+        if new_energy < last_energy:
+            return new_energy + self.max_energy_range_uj - last_energy
+        return new_energy - last_energy
+
+
+class RAPLCPU(cpu_common.CPU):
+    """Control a Single Intel CPU using RAPL interface."""
```

Comment:
```suggestion
    """Control a single CPU that supports RAPL."""
```
RAPL isn't specific to Intel. Try to see if there are other similar mistakes.